primitives/avn-common/src/lib.rs

```diff
@@ -239,19 +239,41 @@ pub fn hash_with_ethereum_prefix(hex_message: &String) -> Result<[u8; 32], ECDSA
     Ok(hash)
 }
 
-pub fn verify_signature<Signature: Member + Verify + TypeInfo, AccountId: Member>(
+pub fn verify_signature<Signature, AccountId>(
     proof: &Proof<Signature, <<Signature as Verify>::Signer as IdentifyAccount>::AccountId>,
     signed_payload: &[u8],
-) -> Result<(), ()> {
+) -> Result<(), ()>
+where
+    Signature: Member + Verify + TypeInfo + codec::Encode,
+    AccountId: Member + codec::Encode + PartialEq,
+    <<Signature as Verify>::Signer as IdentifyAccount>::AccountId: Into<AccountId> + Clone,
+{
     let wrapped_signed_payload: Vec<u8> =
         [OPEN_BYTES_TAG, signed_payload, CLOSE_BYTES_TAG].concat();
-    match proof.signature.verify(&*wrapped_signed_payload, &proof.signer) {
-        true => Ok(()),
-        false => match proof.signature.verify(signed_payload, &proof.signer) {
-            true => Ok(()),
-            false => Err(()),
-        },
+
+    if proof.signature.verify(&*wrapped_signed_payload, &proof.signer) ||
+        proof.signature.verify(signed_payload, &proof.signer)
+    {
+        return Ok(())
     }
+
+    let encoded_signature = proof.signature.encode();
+    if let Ok(ecdsa_signature) = ecdsa::Signature::try_from(&encoded_signature[..]) {
+        let signed_payload_str = String::from_utf8_lossy(signed_payload).to_string();
+
+        if let Ok(ecdsa_pub_key) =
+            recover_public_key_from_ecdsa_signature(&ecdsa_signature, &signed_payload_str)
+        {
+            let hashed_ecdsa_pub_key = keccak_256(ecdsa_pub_key.as_ref());
+            let derived_account = sr25519::Public::from_raw(hashed_ecdsa_pub_key);
```

Comment:
```
	/// A new instance from the given 32-byte `data`.
	///
	/// NOTE: No checking goes on to ensure this is a real public key. Only use it if
	/// you are certain that the array actually is a pubkey. GIGO!
	pub fn from_raw(data: [u8; 32]) -> Self {
		Public(data)
	}
```
The hash of the public key is not guaranteed to be a public key. Maybe instead we could try to use the actual public key (32 bytes version) that we have recovered